mirrord/layer/src/debugger_ports.rs

```diff
@@ -194,23 +264,28 @@ impl DebuggerPorts {
     ///
     /// Log errors (like malformed env variables) but do not panic.
     pub fn from_env() -> Self {
-        let detected = env::var(MIRRORD_DETECT_DEBUGGER_PORT_ENV)
-            .ok()
-            .and_then(|s| {
-                DebuggerType::from_str(&s)
-                    .inspect_err(|e| {
-                        error!(
-                            "Failed to decode debugger type from {} env variable: {}",
-                            MIRRORD_DETECT_DEBUGGER_PORT_ENV, e
-                        )
-                    })
-                    .ok()
-            })
-            .and_then(|d| d.get_port(&std::env::args().collect::<Vec<_>>()));
-        if let Some(port) = detected {
-            env::set_var(MIRRORD_IGNORE_DEBUGGER_PORTS_ENV, port.to_string());
+        let detected: Vec<u16> =
+            match env::var(MIRRORD_DETECT_DEBUGGER_PORT_ENV)
+                .ok()
+                .and_then(|s| {
+                    DebuggerType::from_str(&s)
+                        .inspect_err(|e| {
+                            error!(
+                                "Failed to decode debugger type from {} env variable: {}",
+                                MIRRORD_DETECT_DEBUGGER_PORT_ENV, e
+                            )
+                        })
+                        .ok()
+                }) {
+                Some(debugger) => debugger.get_port(&std::env::args().collect::<Vec<_>>()),
+                None => vec![],
+            };
+        if !detected.is_empty() {
+            detected
+                .iter()
+                .for_each(|port| env::set_var(MIRRORD_IGNORE_DEBUGGER_PORTS_ENV, port.to_string()));
```

Comment:
Doing this in a loop will just overwrite the env. You'll end up with `MIRRORD_IGNORE_DEBUGGER_PORTS=<last_port>` in your env

Reply:
why do we set this var anyway if we return `Self::Detected(detected)` straight after? when would that value get read?

Reply:
In children processes. They have different command line, so we wouldn't be able to detect the port